host-controller/src/main/java/org/jboss/as/domain/controller/operations/HostProcessReloadHandler.java

```diff
@@ -76,10 +76,13 @@
             .setAlternatives(ModelDescriptionConstants.USE_CURRENT_DOMAIN_CONFIG)
             .build();
 
+    protected static final AttributeDefinition TIMEOUT = new SimpleAttributeDefinitionBuilder(ModelDescriptionConstants.TIMEOUT, ModelType.INT, true)
+            .setDefaultValue(new ModelNode(0)).build();
```

Comment:
Sorry, in fact it was an oversight from my side don't comment it here. My initial assumption was wrong. The suspend operation doesn't block for a server shutdown, it was my mistake in my previous comment. For that reason I applied some additional changes which are in commit 8322b20 to allow the new timeout in the reload operation working fine with its suspend operation.

Relevant lines about why the server reloads are not blocking due to a previous suspend executed in other operation are:
GracefulShutdownService L82 --> awaitSuspend method needs the shuttingDown flag as true
DomainServerMain L171 --> at this point the shutdown waits until all activities has been finished or until a time out.

That means if you execute these operation with the code in master branch:
`/host=master/server-config=server-two:suspend(timeout=120)`
`/host=master:reload`
The reload won't block